### PR TITLE
Generic SWxSOC Integration

### PR DESCRIPTION
**Pull request by Alrobbertz**
The goal of this PR is to separate all non-HERMES-specific functionality of the `hermes_core` package to a new generic multi-mission core package. This new multi-mission core package can then be used to share common functionality including CDF file writing, metadata derivations, and schema validation, with other Missions code bases. 

This was partially inspired by discussions at DASH, at the PyHC presentation, at DASH, and my own browsing of other missions that would need to use functionality similar to that in `hermes_core`. 

The main impacts are:
- I/O File Handling completely migrated to generic package
- ISTP required schema derivation and validation migrated to generic package
    - Hermes-specific schema updated here
- Data class functionality moved to generic package

### CODE REVIEW

File: pyproject.toml
Comment (Alrobbertz) [inline pyproject.toml:30]: NOTE: Temp branch pinned to test changes here before they're merged in https://github.com/swxsoc/swxsoc/pull/12

Comment (Alrobbertz) [reply pyproject.toml]: Updated!

File: docs/user-guide/customization.rst
Comment (dbarrous) [inline docs/user-guide/customization.rst:10]: Is it possible to link to the swxsoc docs directly? https://hermes-core--122.org.readthedocs.build/en/122/user-guide/customization.html It looks like it doesn't link via this build

File: docs/user-guide/reading_writing_data.rst
Comment (dbarrous) [inline docs/user-guide/reading_writing_data.rst:577]: Same here, I think you will need to link to the docs page for `swxsoc.util.validation`

File: pyproject.toml
Comment (dbarrous) [inline pyproject.toml:30]:
```suggestion
  'swxsoc @ git+https://github.com/swxsoc/swxsoc@main',
```
Quick change just to ensure it has the same dependency as when referenced in the pipeline

**Review by dbarrous — CHANGES_REQUESTED** [review]
Other than the changes below, this looks good! Will approve when fixed

File: docs/user-guide/reading_writing_data.rst
Comment (Alrobbertz) [reply docs/user-guide/reading_writing_data.rst:577]: ^ It looks like this one does link to the `swxsoc` docs. 
https://hermes-core--122.org.readthedocs.build/en/122/user-guide/reading_writing_data.html#validating-a-cdf-file

File: docs/user-guide/customization.rst
Comment (Alrobbertz) [reply docs/user-guide/customization.rst:10]: It looks like the hermes-core readthedocs can link to the swxsoc docs fine, just that the `config` module is not part of the `swxsoc` api docs. I'm going to make a PR there to update the API docs and then this link should work.

Comment (Alrobbertz) [reply docs/user-guide/customization.rst:10]: https://github.com/swxsoc/swxsoc/pull/16

**Review by dbarrous — APPROVED** [review]
LGTM after the docs changes were made. Thanks!

**Pull request merged**
